### PR TITLE
asc: enable avx2 code path

### PR DESCRIPTION
**Pull request by dvrogozh**
Fixes: #2048

Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>

### CODE REVIEW

File: _studio/shared/asc/CMakeLists.txt [hunk 1]
```diff
@@ -28,14 +28,22 @@ set( defs "" )
...
+
 list( APPEND sources
     ${CMAKE_CURRENT_SOURCE_DIR}/src/asc.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/asc_avx2_impl.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/src/asc_c_impl.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/asc_sse4_impl.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/src/iofunctions.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/src/motion_estimation_engine.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/src/tree.cpp
+    $<TARGET_OBJECTS:asc_avx2>
```
Comment (onabiull) [inline _studio/shared/asc/CMakeLists.txt:45]: Object libraries should go to the LIBS or LIBS_NOVARIANT variables. Please move.

Comment (dvrogozh) [reply _studio/shared/asc/CMakeLists.txt:45]: See here: https://cmake.org/cmake/help/v3.5/command/add_library.html#object-libraries

Comment (dvrogozh) [reply _studio/shared/asc/CMakeLists.txt:45]: @onabiull : do you consider this resolved or you still think we need some change?

Comment (onabiull) [reply _studio/shared/asc/CMakeLists.txt:45]: My bad :)

**Pull request merged**